packages/std/src/query.rs

```diff
@@ -45,10 +45,57 @@ pub enum QueryRequest {
     Balance {
         address: HumanAddr,
     },
+    #[cfg(feature = "staking")]
+    Staking(StakingRequest),
 }
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub struct BalanceResponse {
     pub amount: Option<Vec<Coin>>,
 }
+
+#[cfg(feature = "staking")]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum StakingRequest {
+    Validators {},
+    // Delegations will return all delegations by the delegator,
+    // or just those to the given validator (if set)
+    Delegations {
+        delegator: HumanAddr,
+        validator: Option<HumanAddr>,
+    },
+}
+
+#[cfg(feature = "staking")]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+/// ValidatorsResponse is data format returned from StakingRequest::Validators query
+pub struct ValidatorsResponse {
+    pub validators: Option<Vec<Validator>>,
+}
+
+#[cfg(feature = "staking")]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Validator {
+    pub address: HumanAddr,
+    // TODO: what other info do we want to expose?
```

Comment:
How about adding commission related infos?

In order to let the contract chose proper validator with low commission

Reply:
Good point. What info is needed? And are these relevant in other PoS chains? Or is there a better abstraction?

Is there also some info on if they have been slashed, etc? I would actually defer that to another contract that tracks reputation (via oracle or some complex algorithm) rather than put that info in the blockchain (which cannot be easily customized once launched).

```rust
pub struct Validator {
  // all commision info is measured in 1/100 of percentages, valid values are 0 - 10_000
  pub commission u32,
  pub min_commission u32,
  pub max_commission u32,
}
```

Reply:
sdk defines like this
```
type CommissionRates struct {
	Rate          github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,1,opt,name=rate,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"rate"`
	MaxRate       github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,2,opt,name=max_rate,json=maxRate,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"max_rate" yaml:"max_rate"`
	MaxChangeRate github_com_cosmos_cosmos_sdk_types.Dec `protobuf:"bytes,3,opt,name=max_change_rate,json=maxChangeRate,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Dec" json:"max_change_rate" yaml:"max_change_rate"`
}
```
Maybe we can use this data structure for abstraction 

ref: [Here](https://github.com/cosmos/cosmos-sdk/blob/dfbc6cd81b47b5fa7d49d9174d613cc63b0ea36e/x/staking/types/types.pb.go#L431)